launcher: support Terminal=true

MerlijnWajer opened the pull request:
When an application specifies `Terminal=true`, we want to launch it via `osso-xterm`.

This fixes application launchers like "Vim", "htop", "ncmpcpp" and many more.

MerlijnWajer committed:
launcher: support Terminal=true

We default to x-terminal-emulator, which should default on
osso-xterm.launch on Maemo Leste systems.

For other distributions packaging osso-xterm and no x-terminal-emulator,
they can decide if they want to check for the $TERMINAL environment
variable, or something else.

IMbackK commented:
x-terminal-emulator is nice but this needs a fall though to osso-xterm at the very least, otherwise this feature will only work on debian and debian based distros. I would like to also see $TERMINAL support.

parazyd committed:
launcher: Support fallback to osso-xterm.

This expands on the previous commit, and falls through to osso-xterm if
$TERMINAL is unset, and x-terminal-emulator is not found in $PATH.
